render/egl.c

```diff
@@ -177,10 +166,50 @@ bool wlr_egl_init(struct wlr_egl *egl, EGLenum platform, void *remote_display,
 	egl->egl_exts.dmabuf_import_modifiers =
 		check_egl_ext(egl->exts_str, "EGL_EXT_image_dma_buf_import_modifiers")
 		&& eglQueryDmaBufFormatsEXT && eglQueryDmaBufModifiersEXT;
+	print_dmabuf_formats(egl);
 
 	egl->egl_exts.bind_wayland_display =
 		check_egl_ext(egl->exts_str, "EGL_WL_bind_wayland_display");
-	print_dmabuf_formats(egl);
+
+	egl->egl_exts.context_priority =
+		check_egl_ext(egl->exts_str, "EGL_IMG_context_priority");
+
+	size_t atti = 0;
+	EGLint attribs[5];
+	attribs[atti++] = EGL_CONTEXT_CLIENT_VERSION;
+	attribs[atti++] = 2;
+
+	// Try to reschedule all of our rendering to be completed first. If it
+	// fails, it will fallback to the default priority (MEDIUM).
+	if (egl->egl_exts.context_priority) {
+		attribs[atti++] = EGL_CONTEXT_PRIORITY_LEVEL_IMG;
+		attribs[atti++] = EGL_CONTEXT_PRIORITY_HIGH_IMG;
+	}
```

Comment:
I'd rather see constants and the `EGL_NONE` in `if {} else {}` than this weird `atti++` with the assert here.

Unless there's a good reason to expect that this will be expanded in the future of course